.github/workflows/CI.yml

```diff
@@ -22,9 +22,7 @@ jobs:
           brew ls --versions cmake && brew upgrade cmake || brew install cmake
       - name: Build
         run: | 
-          mkdir -p build
-          cd build
-          cmake ..
-          make
-          sudo make install
+          cmake -B build -DBUILD_SHARED_LIBS=TRUE
```

Comment:
Hum, this forces an always shared build. I think it would make sense to have a static build also, so that it is at least tested. There are some different options needed for static builds and things may not compile/link correctly.

Reply:
Note that before it only build it shared as well. I'm fine with adding a static build too though

Reply:
Yes, I know it was building only shared before - I am not saying you broke it :)

Maybe we can leave it as it is, if someone really needs static build for something (it is rare) and things break there, I am sure they will come and yell here.

Reply:
It seems the CI has passed on this including Mac, so I would be OK with merging your changes. Your changes shouldn't affect Windows, so I would be willing to risk it. I sadly don't have means to build this on Windows neither.